euth/documents/models.py

```diff
@@ -31,6 +31,14 @@ def clean(self, *args, **kwargs):
     def paragraphs_sorted(self):
         return self.paragraphs.all().order_by('weight')
 
+    @cached_property
```

Comment:
Do we like to solve this in a comperable way to how it has been solved with the ideas models (e.g `GenericRelation`)?